### DISCUSSION TITLE
New features

### DISCUSSION THREAD

**Discussion by manducagiuseppe** [bionic-gpt/bionic-gpt#225] [Ideas]
Congratulations on the project, I've tried many in recent months but Bionic turns out to be the best in circulation.
Trying different frameworks of this kind, each one has a functionality that the others lack. I would like to give a piece of advice if I may, based on my experience to make Bionic truly unbeatable.
I would add two features that I think are very important.

1. The first feature is the ability to add URLs of websites as documents (I found this feature very interesting on https://github.com/cheshire-cat-ai). 
2. The second feature I would add is to have the assistant cite the file it draws from as the source of its response.

Thanks again for your work.

**Comment by ianpurton** [comment]
> ability to add URLs of websites as documents

Does it then download the webpage, or do you mean a URL to a document and it downloads the document?

**Comment by manducagiuseppe** [comment]
Reads the text or download the pages and stores it in memory as if it were a document
